framework/include/vectorpostprocessors/MaterialVectorPostprocessor.h

```diff
@@ -0,0 +1,52 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef MATERIALVECTORPOSTPROCESSOR_H
+#define MATERIALVECTORPOSTPROCESSOR_H
+
+#include "ElementVectorPostprocessor.h"
+
+//Forward Declarations
+class MaterialVectorPostprocessor;
+
+template<>
+InputParameters validParams<MaterialVectorPostprocessor>();
+
+/// This postprocessor records all scalar material properties of the specified
+/// material object on specified elements at the indicated execution points
+/// (e.g. initial, timestep_begin, etc.).  Non-scalar properties are ignored
+/// with a warning.
+class MaterialVectorPostprocessor :
+  public ElementVectorPostprocessor
+{
+public:
+  MaterialVectorPostprocessor(const InputParameters & parameters);
+
+  virtual void initialize() override { };
+  virtual void execute() override;
+  virtual void finalize() override;
+  virtual void threadJoin(const UserObject & y) override;
+
+private:
+  void sortVecs();
+
+  std::set<unsigned int> _elem_filter;
```

Comment:
Can you put Doxygen style `///` comments for each of these, if there is a bunch of stuff that is similar you can used a single comment and wrap with `\\\@{` and `\\\@}`